### PR TITLE
Add a how to review section to dev docs

### PR DESCRIPTION
**Pull request by tbekolay**
**Description:**
This adds a "Reviewing" section to the docs that documents how we expect reviews to happen in Nengo projects.

**Motivation and context:**
This text is my first pass trying to write down the general steps I take when reviewing. Hopefully by writing it down we can convert more developers into reviewers.

**How long should this take to review?**

- Average (neither quick nor lengthy)

**Types of changes:**

- Documentation

**Checklist:**

- [x] I have read the **CONTRIBUTING.rst** document.
- [x] I have updated the documentation accordingly.
- [ ] I have included a changelog entry.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

**Still to do:**

- [x] Discuss this document at the reviewing workshop on June 15, 2:30pm.
- [x] Incorporate changes from the workshop.

### CODE REVIEW

**Comment by tbekolay** [comment]
I've pushed a commit to incorporate the discussion from the reviewing workshop. This should be ready for review; please anyone that was at the reviewing workshop take a look at this and make sure I haven't missed or misrepresented anything!

File: docs/contributing.rst [hunk 1]
```diff
@@ -415,9 +488,21 @@ Your decision should be one of the following:
 1. This PR is good to merge, or will be good to merge with my changes.
 2. This PR could be good to merge, but it requires significant changes
    that I am working on.
-3. This PR is not apporpriate for this project.
-
-Note that the third option should not be taken lightly,
+3. This PR could be good to merge, but it requires significant changes.
+4. This PR is not apporpriate for this project.
```
Comment (drasmuss) [inline docs/contributing.rst]: appropriate (fix in merge)

File: docs/contributing.rst [hunk 2]
```diff
@@ -271,13 +271,20 @@ Here are the steps developers should take when doing a review.
 
 1. Assign yourself to the PR to let others know you're reviewing it.
 2. Familiarize yourself with the part of the codebase that the PR changes.
-3. Read through the code changes.
+3. Read through the code changes and commit messages.
 4. Test the PR branch.
-5. Make inline comments for minor changes.
-6. Make FIXUP commits for other changes.
+5. Make inline comments.
+6. Make commits for other changes.
```
Comment (drasmuss) [inline docs/contributing.rst]: Maybe "proposed changes" instead of "other changes"? "other changes" just reads kind of ambiguously to me.

**Comment by drasmuss** [comment]
Looks good to me.  I realize I made inline comments instead of commits, which is one of the main things we're addressing here, but they were both so minor I'd just do them in the merge.

**Comment by tbekolay** [comment]
S'aight ;) I realize that my spellcheck isn't turned on atm so I should do that anyhow...

**Comment by jgosmann** [comment]
LGTM, +1 on @drasmuss  comments.
Minor question: Should we write `fixup!` and `squash!` commits as opposed to `fixup` and `squash` commits because that corresponds to what is prependend to the commit message?

**Comment by drasmuss** [comment]
Just occurred to me that it'd be nice if we had a tag to indicate that the PR has been reviewed once.  Something in between "needs review" and "reviewed".  It'd help us track the status of a PR, and provide some way for the first reviewer to officially indicate that they have signed off on the PR.

So a PR would start off as "needs review", after one reviewer signs off we'd change the status to "needs second review", and then when the second reviewer signs off we'd change the status to "reviewed".

**Comment by hunse** [comment]
> Just occurred to me that it'd be nice if we had a tag to indicate that the PR has been reviewed once. Something in between "needs review" and "reviewed".

I assumed that the merger was always doing the second review, so whenever I've reviewed something but I'm not going to be the one merging it, then I mark it "reviewed". If I review it and I am going to be merging it, then I give feedback and say whether it looks good or not, but I don't change the label.

**Comment by tbekolay** [comment]
I think it's confusing for ordinary reviewers and maintainer reviewers to do different things... it's entirely possible to get two reviews from non-maintainers, so I would support something that is more straightforward regardless of whether the reviewer is a maintainer or not. I like the labels `needs 2 reviews` and `needs 1 review`, that seems the most explicit to me, but I'm okay with "needs second review" if that's the consensus. In either case, I think we should do something, which I'll note in this PR before it's merged!

**Comment by jgosmann** [comment]
> I like the labels needs 2 reviews and needs 1 review

With this it might not clear to a new comer which of the two labels to assign to PR at first. `needs review` and `needs second review` seems to be clearer in this respect.

**Comment by tbekolay** [comment]
> might not clear to a new comer which of the two labels to assign to PR at first

Fair point! I'm swayed to a `needs second review` label.

**Comment by tbekolay** [comment]
Added a commit explaining how we do labels, which includes the 'needs second review' label.

**Comment by Seanny123** [comment]
LGTM.

File: docs/contributing.rst [hunk 3]
```diff
@@ -253,6 +253,310 @@ Commit messages should fit the following template.
...
+   ``git commit --squash <commit hash>`` where the commit hash
+   corresponds to the commit that your ``squash`` commit modifies.
+   Unlike with the ``--fixup`` option, git will now prompt you
+   to enter a message to explain what your ``squash`` commit does.
+
+   ``squash`` commits are so named because
+   the maintainer will ``squash`` those commits into
+   the appropriate part of the PR branch's history
+   before merging that branch into ``master``.
+   Since ``squash`` commits contain a commit message,
```
Comment (drasmuss) [inline docs/contributing.rst]: Just noticed this when getting ready to merge, looks like the end of this sentence disappeared somewhere

Comment (tbekolay) [inline docs/contributing.rst]: I think it never existed in the first place... will finish my thought!

**Comment by tbekolay** [comment]
Sorry for the delay; I've updated this now and squashed all the commit, so should ready for a last look-over and merge.

**Comment by Seanny123** [comment]
LGTM. Please merge it.

**Pull request merged**